File: Tipos de Dados/Primitivos/Primitivas.pas
//Pascal
program Primitivas;
var
    inteiroPequeno: shortint;      // Inteiro pequeno
    inteiro: integer;              // Inteiro
    inteiroGrande: int64;          // Inteiro grande
    pontoFlutuanteSimples: single; // Ponto flutuante simples
    pontoFlutuanteDuplo: double;   // Ponto flutuante de dupla precisão
    booleano: boolean;             // Booleano
    caractere: char;               // Caractere
    inteiroSemSinal: word;         // Inteiro sem sinal
begin
    inteiroPequeno := 100;
    inteiro := 1000;
    inteiroGrande := 1000000;
    pontoFlutuanteSimples := 3.14;
    pontoFlutuanteDuplo := 3.14159265359;
    booleano := true;
    caractere := 'A';
    inteiroSemSinal := 2000;

    writeln('Inteiro pequeno: ', inteiroPequeno);
    writeln('Inteiro: ', inteiro);
    writeln('Inteiro grande: ', inteiroGrande);
    writeln('Ponto flutuante simples: ', pontoFlutuanteSimples:0:2);
    writeln('Ponto flutuante de dupla precisão: ', pontoFlutuanteDuplo:0:10);
    writeln('Booleano: ', booleano);
    writeln('Caractere: ', caractere);
    writeln('Inteiro sem sinal: ', inteiroSemSinal);
end.

File: Tipos de Dados/Primitivos/primitivas.kt
//Kotlin
fun main() {
    val pequeno: Byte = 127              // Inteiro pequeno
    val inteiro: Int = 2147483647         // Inteiro padrão
    val grande: Long = 9223372036854775807 // Inteiro grande
    val pontoFlutuante: Float = 3.14f     // Ponto flutuante simples
    val duplaPrecisao: Double = 3.1415926535 // Ponto flutuante de dupla precisão
    val booleano: Boolean = true          // Booleano
    val caractere: Char = 'K'             // Caractere
    val inteiroSemSinal: UInt = 4294967295U // Inteiro sem sinal (Kotlin suporta inteiros sem sinal)

    println("Byte: $pequeno")
    println("Int: $inteiro")
    println("Long: $grande")
    println("Float: $pontoFlutuante")
    println("Double: $duplaPrecisao")
    println("Boolean: $booleano")
    println("Char: $caractere")
    println("UInt: $inteiroSemSinal")
}
